src/OpenSearch/Endpoints/AbstractEndpoint.php

```diff
@@ -280,4 +282,39 @@ private function isNestedArray(array $a): bool
 
         return false;
     }
+
+    /**
+     * This function returns all param deprecations also optional with an replacement field
+     *
+     * @return array<string, string|null>
+     */
+    protected function getParamDeprecation(): array
+    {
+        return [];
+    }
+
+    private function checkForDeprecations(): void
+    {
+        $deprecations = $this->getParamDeprecation();
+
+        if ($deprecations === []) {
+            return;
+        }
+
+        $keys = array_keys($this->params);
+
+        foreach ($keys as $key) {
```

Comment:
Can we use `array_intersect` and throw the error for the entire set of parameters? Reduces the amount of errors and makes it easier to debug in 1 go instead of having to fix parameter by parameter, and try again.

Reply:
When we merge them, we can't say anymore what's the replacement 🤔

Reply:
@dblock is that fine?

Reply:
I think so